### PR TITLE
Fix download ninja and download directory

### LINKED ISSUE #3708: [BUG] Three issues with build_ninja.bash
**Describe the bug**
1) https://github.com/OpenImageIO/oiio/blob/master/src/build-scripts/build_ninja.bash#L13 performs a check to see if a .zip file exists.  If it doesn't a .tar.gz file is downloaded.  Either way a .tar.gz file is extracted.  The check should be for the existance of a .tar.gz file, not a .zip file.
2) The path of the file downloaded https://github.com/ninja-build/ninja/archive/v1.10.2.tar.gz, no longer aligned with GitHub's release file structure.  The download is now available at https://github.com/ninja-build/ninja/archive/refs/tags/v1.10.2.tar.gz.  "refs/tags" needs to be added.

**To Reproduce**
1. Run src/build-scripts/build_ninja.bash
2. Or run src/build-scripts/build_ninja.bash with oiio/ext/downloads/ninja-1.10.2.zip already present.  You can download it from https://github.com/ninja-build/ninja/archive/refs/tags/v1.10.2.zip

**Expected behavior**
If the .zip file is present, but the .tar.gz file is not, the script will attempt to untar the nonexistent file and fail.  If neither file is present, the script will attempt to download the .tar.gz file file from the wrong location and fail.

**Evidence**
```
+ LOCAL_DEPS_DIR=/home/cent/git/external/oiio/ext
+ DOWNLOADS_DIR=/home/cent/git/external/oiio/ext/downloads
+ NINJA_REPO=https://github.com/ninja-build/ninja
+ NINJA_VERSION=1.10.2
+ NINJA_BRANCH=v1.10.2
+ NINJA_INSTALL_DIR=/home/cent/git/external/oiio/ext/dist/bin
+ '[' '!' -f /home/cent/git/external/oiio/ext/downloads/ninja-1.10.2.zip ']'
+ curl --location https://github.com/ninja-build/ninja/archive/v1.10.2.tar.gz -o /home/cent/git/external/oiio/ext/downloads/ninja-v1.10.2.tar.gz
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0Warning: Failed to create the file
Warning: /home/cent/git/external/oiio/ext/downloads/ninja-v1.10.2.tar.gz: No
Warning: such file or directory
  0  208k    0   512    0     0    773      0  0:04:36 --:--:--  0:04:36  2327
curl: (23) Failure writing output to destination
```

**Platform information:**
 - OIIO branch/version: commit 1449b465000bf7c83e983cf870ca7b1f2966d446 (HEAD -> master, origin/master, origin/HEAD)
 - OS: CentOS Stream release 9
 - C++ compiler:  g++ 11.3.1 20221121
 - Any non-default build flags when you build OIIO:  No.  Not getting that far.

### PR DESCRIPTION
**Pull request by hisergiorojas**
## Description

Fix #3708
## Checklist:

- [ ] I have read the [contribution guidelines](https://github.com/AcademySoftwareFoundation/OpenImageIO/blob/master/CONTRIBUTING.md).
- [ ] I have updated the documentation, if applicable.
- [ ] I have ensured that the change is tested somewhere in the testsuite
  (adding new test cases if necessary).
- [ ] If I added or modified a C++ API call, I have also amended the
  corresponding Python bindings (and if altering ImageBufAlgo functions, also
  exposed the new functionality as oiiotool options).
- [ ] My code follows the prevailing code style of this project. If I haven't
  already run clang-format before submitting, I definitely will look at the CI
  test that runs clang-format and fix anything that it highlights as being
  nonconforming.

### CODE REVIEW

File: src/build-scripts/build_ninja.bash [hunk 1]
```diff
@@ -14,8 +14,18 @@ NINJA_VERSION=${NINJA_VERSION:=1.10.2}
 NINJA_BRANCH=${NINJA_BRANCH:=v${NINJA_VERSION}}
 NINJA_INSTALL_DIR=${NINJA_INSTALL_DIR:=${LOCAL_DEPS_DIR}/dist/bin}
 
-if [ ! -f $DOWNLOADS_DIR/ninja-${NINJA_VERSION}.zip ]; then
-    curl --location ${NINJA_REPO}/archive/${NINJA_BRANCH}.tar.gz -o $DOWNLOADS_DIR/ninja-${NINJA_BRANCH}.tar.gz
+# Check if the directories exist, create them if not
+if [ ! -d "$LOCAL_DEPS_DIR" ]; then
+    mkdir -p "$LOCAL_DEPS_DIR"
+fi
```
Comment (lgritz) [inline src/build-scripts/build_ninja.bash:20]: I think you don't need the `if` here: just the `mkdir -p` will do it, and the -p flag makes it not be an error if the directory already exists.

**Comment by lgritz** [comment]
Is there a reason this is marked as a "draft"? It looks right to me and I'm happy to merge it. (I did point out a small possible simplification, though it is correct as-is.)

**Comment by hisergiorojas** [comment]
When I did the draft pull request, I just needed some time to read the contribution guidelines and get my signed CLA-approved

File: src/build-scripts/build_ninja.bash (hunk 1 above)
Comment (lgritz) [reply src/build-scripts/build_ninja.bash:20]: I think Lee is right. The `mkdir -p "$DOWNLOADS_DIR"` should do the whole job, no?

Comment (hisergiorojas) [reply src/build-scripts/build_ninja.bash:20]: Yes, it was to able create the whole download directory.

**Review by lgritz — APPROVED** [review]

**Pull request merged**